### PR TITLE
Move slow tests to nightly

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
Move certain slow tests to nightly. The following modules are not present in this [list of important pipelines](https://github.com/huggingface/diffusers/pull/4431#issuecomment-1662018064) and are some of the slowest tests in the CI/CD pipeline. 

```
tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py
tests/pipelines/stable_diffusion/test_stable_diffusion_panorama.py
```

The `test_full_model` test in `tests/pipelines/text_to_video/test_text_to_video.py` is a similar test to `test_two_step_model` except that it runs for a greater number of inference steps. I propose moving the long running test to nightly and keeping the shorter 2 step test in `SlowTests`  

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py [hunk 1]
```diff
@@ -539,47 +539,6 @@ def test_stable_diffusion_2_pix2pix_inversion(self):
 
         assert np.abs(expected_slice - image_slice.cpu().numpy()).max() < 5e-2
 
-    def test_stable_diffusion_pix2pix_full(self):
```
Comment (sayakpaul) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py:542]: Why are we getting rid of it entirely?

File: tests/pipelines/text_to_video/test_text_to_video.py [hunk 2]
```diff
@@ -170,36 +170,42 @@ def test_progress_bar(self):
...
+        pipe = pipe.to(torch_device)
 
         prompt = "Spiderman is surfing"
         generator = torch.Generator(device="cpu").manual_seed(0)
 
-        video_frames = pipe(prompt, generator=generator, num_inference_steps=25, output_type="pt").frames
+        video_frames = pipe(prompt, generator=generator, num_inference_steps=2, output_type="pt").frames
         video = video_frames.cpu().numpy()
 
         assert np.abs(expected_video - video).mean() < 5e-2
```
Comment (sayakpaul) [inline tests/pipelines/text_to_video/test_text_to_video.py:189]: Works for me!

File: tests/pipelines/text_to_video/test_text_to_video.py [hunk 3]
```diff
@@ -170,36 +170,42 @@ def test_progress_bar(self):
...
+        pipe = pipe.to(torch_device)
 
         prompt = "Spiderman is surfing"
         generator = torch.Generator(device="cpu").manual_seed(0)
 
-        video_frames = pipe(prompt, generator=generator, num_inference_steps=2, output_type="pt").frames
+        video_frames = pipe(prompt, generator=generator, num_inference_steps=25, output_type="pt").frames
         video = video_frames.cpu().numpy()
 
         assert np.abs(expected_video - video).mean() < 5e-2
```
Comment (sayakpaul) [inline tests/pipelines/text_to_video/test_text_to_video.py:211]: If the two step model is passing the assertion, I think it's okay to remove it from nightly unless we have a strong reason to.

**Review by sayakpaul — APPROVED** [review]
Sweet!

File: tests/pipelines/text_to_video/test_text_to_video.py (hunk 3 above)
Comment (sayakpaul) [reply tests/pipelines/text_to_video/test_text_to_video.py:211]: Remove the full and keep the two steps.

File: tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py (hunk 1 above)
Comment (DN6) [reply tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py:542]: The pipeline wasn't prioritised in [this list](https://github.com/huggingface/diffusers/issues/4527#issue-1840665874) (under highly important and important) 

So I moved all the slow tests to nightly.

**Pull request merged**